### PR TITLE
refactor(users): Update Database connection for Read only functions

### PR DESCRIPTION
**Pull request by zeel991**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
This PR refactors the database connection logic by changing the function from `pg_connection_write()` to `pg_connection_read()` for functions that only perform read operations from the database. This is an internal change aimed at optimizing resource usage. All user-facing APIs should continue to function as expected without any disruptions.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
This change is required to improve efficiency by ensuring that we use the appropriate connection type for read operations. It resolves the inefficiency of using write connections for read-only database queries. 

- Fixes https://github.com/juspay/hyperswitch/issues/5965

## How did you test it?
- Verified that all user-related APIs are working without any issues.
- All existing test cases for database read and write operations passed successfully.
- Confirmed that this internal change does not affect the functionality of user-facing APIs.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [x] I added unit tests for my changes where possible

### CODE REVIEW

**Review by ThisIsMani — CHANGES_REQUESTED** [review]
Please change only the files mentioned in the issue.

File: crates/router/src/db/user.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ impl UserInterface for Store {
         &self,
         user_data: storage::UserNew,
     ) -> CustomResult<storage::User, errors::StorageError> {
-        let conn = connection::pg_connection_write(self).await?;
+        let conn = connection::pg_connection_read(self).await?;
```
Comment (ThisIsMani) [inline crates/router/src/db/user.rs:62]: This shouldn't be read.

**Review by ThisIsMani — CHANGES_REQUESTED** [review]

**Review by ThisIsMani — APPROVED** [review]

**Comment by ThisIsMani** [comment]
Please add something in description saying that this is a internal changes and all the user related APIs should work without any issues in the testcases section.

**Comment by zeel991** [comment]
> Please add something in description saying that this is a internal change and all the user related APIs should work without any issues in the testcases section.

Done @ThisIsMani , I apologize for any beginner mistakes I made while raising this PR. Thank you for your patience and support

**Review by SanchithHegde — APPROVED** [review]

**Comment by ThisIsMani** [comment]
Hey @zeel991, can you check the original template and change you PR description into that template.

**Comment by zeel991** [comment]
> Hey @zeel991, can you check the original template and change you PR description into that template.

Are these changes appropriate ?

**Comment by ThisIsMani** [comment]
Yeah, looks fine.

**Comment by zeel991** [comment]
Hello @ThisIsMani ,would you please consider adding the `hactoberfest` tag to the PR

**Comment by zeel991** [comment]
Hey @ThisIsMani , since this PR is approved, when can it be merged ?

**Comment by ThisIsMani** [comment]
@zeel991, will get this merged by Monday.

**Comment by ThisIsMani** [comment]
Hey @zeel991, can you resolve the conflicts. I will get this merged as soon as you resolve.

**Comment by zeel991** [comment]
> Hey @zeel991, can you resolve the conflicts.

Done ! 👍🏻

**Comment by zeel991** [comment]
Hello @ThisIsMani ,
I am Zeel Darji , C4GT community Lead and Open source Lead of Development Wing of IIITN . I am organizing an open source event in my college's techfest which higlights the importance of DPGs and DPIs .

I wanted to ask you if participants could contribute to the repos of Juspay for our Event ?

**Comment by gorakhnathy7** [comment]
Hey @zeel991,
Happy to help with the event.
Pinged you in community workspace, we can continue the discussion there.

**Comment by zeel991** [comment]
> Hey @zeel991, Happy to help with the event. Pinged you in community workspace, we can continue the discussion there.

Can you please send the link , I couldn't find it

**Comment by gorakhnathy7** [comment]
> > Hey @zeel991, Happy to help with the event. Pinged you in community workspace, we can continue the discussion there.
> 
> Can you please send the link , I couldn't find it

Please check [Hyperswitch Slack workspace
](https://join.slack.com/t/hyperswitch-io/shared_invite/zt-2jqxmpsbm-WXUENx022HjNEy~Ark7Orw)

**Review by ThisIsMani — APPROVED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Comment by zeel991** [comment]
@ThisIsMani Should I update the branch ?

**Comment by ThisIsMani** [comment]
But seems like formatting check if failing in the PR, can you format the code using `just fmt` and push the changes.

**Comment by zeel991** [comment]
> But seems like formatting check if failing in the PR, can you format the code using `just fmt` and push the changes.

I have removed the merge conflict and formatted the PR

**Comment by zeel991** [comment]
Hey @ThisIsMani If my contribution is as required , can you please merge my PR , as I have been getting conflicts

**Review by ThisIsMani — APPROVED** [review]

**Comment by zeel991** [comment]
@ThisIsMani All the tests have passed

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**